onnx/onnx.in.proto

```diff
@@ -417,12 +514,47 @@ message TypeProto {
 
 // #if ONNX-ML
 
-  // repeated T
-  message Sequence {
-  // The type and optional shape of each element of the sequence.
-  // This field MUST be present for this version of the IR.
-    optional TypeProto elem_type = 1;
-  };
+  message Scalar { 
+    // This field MUST NOT have the value of UNDEFINED 
+    // This field MUST be present for this version of the IR. 
+    optional TensorProto.DataType data_type = 1; 
+  }; 
+
+  message Sequence { 
+    // The type and optional shape of each element of the sequence. 
+    // This field MUST be present for this version of the IR. 
+    optional TypeProto elem_type = 1; 
```

Comment:
Add an "optional length"?

Reply:
It could play a role similar to "shape" in TensorType.

Reply:
It could, but it seems to be an update to an existing type that isn't motivated by a concrete use case.

Reply:
As I was going through the TLC transforms, I noticed several that convert one (column) value into a sequence/vector of values of statically fixed size. If we do code-generation, we can definitely exploit statically-fixed sizes: e.g., for sequence/vectors of small size (4 or 5), loop-unrolling might make sense. If these are encoded as Tensors, we can currently exploit the tensor's shape to do such optimizations. But if these are encoded as Sequences, we lack the same information.